File: 0066-plus-one/0066-plus-one.java
class Solution {
    public int[] plusOne(int[] digits) {
        int val = digits[digits.length - 1];
        if(val < 9){
            digits[digits.length - 1]++;
            return digits;
        }
        else
        {
            boolean gotIt = false;
            for(int i = digits.length - 1; i >= 0 ; --i)
            {
                if(digits[i] == 9 && !gotIt)
                {
                    digits[i] = 0;
                }
                else if(digits[i] != 9 && !gotIt)
                {
                    digits[i]++;
                    gotIt = true;
                }
            }
            if(digits[0] == 0)
            {
                int ansarray[] = new int[digits.length + 1];
                ansarray[0] = 1;
                for(int i = 1 ; i < digits.length ; i++)
                    ansarray[i] = digits[i];
                return ansarray;
            }
            return digits;
            
        }
    }
}

File: 0066-plus-one/0066-plus-one.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        int val = digits[digits.size() - 1];
        if(val < 9)
        {
            digits[digits.size() - 1]++;
            return digits;
        }
        else{
            vector<int> ans;
            bool gotIt = false;
            for(int i = digits.size() - 1 ; i >= 0 ; --i)
            {
                if(digits[i] == 9 and !gotIt)
                {
                    ans.emplace_back(0);
                }
                if(digits[i] != 9 and !gotIt)
                {
                    digits[i]++;
                    gotIt = true;
                }
                if(gotIt)
                    ans.emplace_back(digits[i]);
            }
            if(ans[ans.size() - 1] == 0)
                ans.emplace_back(1);
            reverse(ans.begin(),ans.end());
            return ans;
        }
    }
};